### PR TITLE
SPIKE Inline validation

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-elemental/issues/1068

DO NOT MERGE

This PR is only for investigative purposes

### CODE REVIEW

**Comment by GuySartorelli** [comment]
There's so much going on here it's hard to see what's doing what. Can you please add a few notes in the PR description explaining what all this PR is doing?

**Comment by emteknetnz** [comment]
I don't think this PR is worth peer reviewing, it's not in a good state and it's not complete. It's not meant to be merged, it's kind of just a dumping ground for code I used to do investigation

**Comment by GuySartorelli** [comment]
Is there anything in this PR that's useful for understanding and/or evaluating the high level approach you're recommending in the issue? Currently the issue is more of an info dump than a "this is the recommended approach" so it's a bit hard to follow from that angle.

**Comment by GuySartorelli** [comment]
Closing this for now - we may end up copying some bits from this when we implement this validation in https://github.com/silverstripe/silverstripe-elemental/issues/329 but it likely won't use this exact PR.
